### PR TITLE
Create tags table and model.

### PR DESCRIPTION
**Pull request by abhikanojia**
- Create tags table and model.
- Create seed data for tags.

### CODE REVIEW

File: app/models/tag.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+class Tag < ApplicationRecord
```
Comment (sawangupta92) [inline app/models/tag.rb]: validations?

File: db/seeds.rb [hunk 2]
```diff
@@ -1,7 +1,11 @@
...
+tags_list = [
+  'web development', 'rails', 'rails 5', 'education',
+  'entertainment', 'ruby on rails', 'software engineers',
+  'finance', 'space', 'banks', 'fiction', 'actors',
+  'enterpreneurship', 'places', 'philosophy', 'physics',
+  'religion', 'books', 'travel', 'science', 'computer science'
+]
+
+tags_list.each do |tag|
+ Tag.create(name: tag, slug: tag.parameterize)
```
Comment (sawangupta92) [inline db/seeds.rb]: create!

File: app/models/tag.rb (hunk 1 above)
Comment (abhikanojia) [reply app/models/tag.rb]: Validations will be done in different branch where i Will merge both questions section and tags section.

File: app/models/tag.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+  validates :name, presence: true, uniqueness: { case_sensitive: false }, length: { within: 2..20 }, tag: true
```
Comment (sawangupta92) [inline app/models/tag.rb]: 2..20 should be a constant

File: app/models/tag.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+  before_save :add_slug
```
Comment (sawangupta92) [inline app/models/tag.rb]: we don't need to update the slug.

File: app/validators/tag_validator.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
+class TagValidator < ActiveModel::EachValidator
```
Comment (sawangupta92) [inline app/validators/tag_validator.rb]: rename

**Pull request merged**